core/src/main/java/org/apache/brooklyn/core/config/ConfigUtils.java

```diff
@@ -126,4 +128,20 @@ else if (HasConfigKey.class.isAssignableFrom(f.getType()))
         }
         return Collections.unmodifiableSet(result);
     }
+
+    /**
+     * Look for keys with <code>configPrefix</code> in <code>configBag</code> and
+     * those which have <code>configPrefix</code> are added in <code>destinationBucket</code> but without <code>configPrefix</code>.
+     * @param configPrefix prefix to look for
+     * @param configBag keys to look in
+     * @param destinationBucket should not be an ImmutableMap
+     */
+    public static void addUnprefixedConfigKeyInConfigBack(String configPrefix, ConfigBag configBag, Map<String, Object> destinationBucket) {
+        for (Map.Entry<ConfigKey<?>, ?> entry : configBag.getAllConfigAsConfigKeyMap().entrySet()) {
+            String keyName = entry.getKey().getName();
+            if (keyName.startsWith(configPrefix)) {
+                destinationBucket.put(Strings.removeFromStart(keyName, configPrefix), entry.getValue());
+            }
+        }
+    }
```

Comment:
I made it this way because it is convenient to use this way in https://github.com/apache/brooklyn-server/blob/master/core/src/main/java/org/apache/brooklyn/location/ssh/SshMachineLocation.java#L636-L645